docs/GAME.md

```diff
@@ -0,0 +1,83 @@
+# Game Design Notes
+
+This file contains all design choices for the game logic and gameplay portion of the project. Segments marked `DEV` are in development and are not finalized.
+
+## Game Logic
+
+### Entity-Component-System (ECS) `DEV`
+
+ECS is a architectural pattern that follows a "composition over inheritance" principle.
+
+#### Entity 
+
+A general purpose object identified by a unique id. Every entity type should have its own set of unique identifiers.
+
+#### Component
+
+Raw data to describe an aspect of the object. Typically implemented using a struct or class.
+
+#### System
+
+Each system runs continuously and preforms global actions on every entity that possesses a Component of the same type.
+
+Note: Possibility of implementing multi-threading here.
+
+## Gameplay
+
+### Combat `DEV`
+
+I'm feeling ARPG zelda style but with more combat options.
+
+### Stats
+
+Each statistic should have equal beneficial impact on gameplay. There should not be a statistic that does not directly impact play style. My suggestion for stats:
+
+`Constitution` - Defines the character's health. Tanky builds should have high CON.
+`Defense` - Defines how much damage a character takes. Armored builds should habe high DEF.
```

Comment:
How does this differ from Constitution? Is this like armour?

Reply:
A class with high defense will take less damage, while a class with high constitution will have more HP. It's possible that both these qualities would better serve being treated as one attribute but usually games have them seperate.